### ISSUE TITLE
source needs comments

### ISSUE THREAD

**Issue by ayang64** [chipk/fw_recert#1]
We should document all of the public functions in the code.

**Comment by chipk** [comment]
Yes.  I am working on it.  I have to study some of the bits to get a better understanding of the details. I would appreciate a review from you when I push my changes.

**Comment by ayang64** [comment]
yup -- no problem. i merely opened the bug to track it.  if you like, i recommend submitting a  pull request and i can review it.
